torch/_jit_internal.py

```diff
@@ -717,7 +717,7 @@ def is_rref(ann):
 
 def is_final(ann):
     return ann.__module__ in {'typing', 'typing_extensions'} and \
-        (getattr(ann, '__origin__', None) is Final)
+        (getattr(ann, '__origin__', None) is Final or isinstance(ann, type(Final)))
```

Comment:
Optional:

Can you say a bit more about why this change is needed (in PR description)? I am curious.

Reply:
added :)